crates/consensus/src/transaction/envelope.rs

```diff
@@ -95,64 +103,90 @@ impl TryFrom<u8> for TxType {
 #[non_exhaustive]
 pub enum TxEnvelope {
     /// An untagged [`TxLegacy`].
-    Legacy(Signed<TxLegacy>),
+    Legacy(Enveloped<TxLegacy>),
     /// A [`TxEip2930`] tagged with type 1.
-    Eip2930(Signed<TxEip2930>),
+    Eip2930(Enveloped<TxEip2930>),
     /// A [`TxEip1559`] tagged with type 2.
-    Eip1559(Signed<TxEip1559>),
+    Eip1559(Enveloped<TxEip1559>),
     /// A TxEip4844 tagged with type 3.
     /// An EIP-4844 transaction has two network representations:
     /// 1 - The transaction itself, which is a regular RLP-encoded transaction and used to retrieve
     /// historical transactions..
     ///
     /// 2 - The transaction with a sidecar, which is the form used to
     /// send transactions to the network.
-    Eip4844(Signed<TxEip4844Variant>),
+    Eip4844(Enveloped<TxEip4844Variant>),
     /// A [`TxEip7702`] tagged with type 4.
-    Eip7702(Signed<TxEip7702>),
+    Eip7702(Enveloped<TxEip7702>),
 }
```

Comment:
excellent, **but**, what's the plan for `alloy_consensus::TypedTransaction` now? it's essentially the same type

Reply:
`TypedTransaction` is the unsigned version. As such neither signature or hash is known, and it just includes a signable version of the inner transaction and is used in the `NetworkWallet::<N>::sign_transaction_from` API

actually now that you mention it, we need to apply #1489 to `TypedTransaction` as well

Reply:
#1494